### PR TITLE
Glossary: Add Zero-knowledge proof, STARK and SNARK

### PR DESCRIPTION
**Pull request by mmilenkovic**
## Description

Adding missing glossary terms as previously approved (linked related issues).

## Related Issue

1. [Zero-knowledge proof #5121](https://github.com/ethereum/ethereum-org-website/issues/5121)
2. [STARK #5120](https://github.com/ethereum/ethereum-org-website/issues/5120)
3. [SNARK #5119](https://github.com/ethereum/ethereum-org-website/issues/5119)

### CODE REVIEW

File: src/content/glossary/index.md [hunk 1]
```diff
@@ -681,6 +681,14 @@ A program that executes on the Ethereum computing infrastructure.
   Introduction to Smart Contracts
 </DocLink>
 
+### SNARK {#snark}
+
+Succinct Non-interactive ARgument of Knowledge a type of a [zero-knowledge proof](#zk-proof).
```
Comment (minimalsm) [inline src/content/glossary/index.md:686]:
```suggestion
Short for "succinct non-interactive argument of knowledge", a SNARK is a type of [zero-knowledge proof](#zk-proof).
```

File: src/content/glossary/index.md [hunk 2]
```diff
@@ -713,6 +721,14 @@ Depositing a quantity of [ether](#ether) (your stake) to become a validator and
   Stake your ETH to become an Ethereum validator
 </DocLink>
 
+### STARK {#stark}
+
+Scalable Transparent ARgument of Knowledge a type of a [zero-knowledge proof](#zk-proof).
```
Comment (minimalsm) [inline src/content/glossary/index.md:726]:
```suggestion
Short for "scalable transparent argument of knowledge", a STARK is a type of [zero-knowledge proof](#zk-proof).
```

File: src/content/glossary/index.md [hunk 3]
```diff
@@ -840,6 +856,14 @@ The smallest denomination of [ether](#ether). 10<sup>18</sup> wei = 1 ether.
 
 A special Ethereum address, composed entirely of zeros, that is specified as the destination address of a [contract creation transaction](#contract-creation-transaction).
 
+### Zero-knowledge proof {#zk-proof}
+
+Zero-knowledge proof is a cryptographic method by which one party (the prover) can prove to another party (the verifier) that a given statement is true while the prover avoids conveying any additional information.
```
Comment (minimalsm) [inline src/content/glossary/index.md:861]: I'd be in favour of using simpler language here.

Suggesting: 

1. I don't think we need to specify person A proves to person B. I believe the number of verifiers in a zkp could be arbitrary and we can therefore generalise this statement. (correct me if I'm wrong here)
2. Remove formal language for more colloquial language

```suggestion
Zero-knowledge proof is a cryptographic method that allows an individual to prove that a statement is true without conveying any additional information.
```

File: src/content/glossary/index.md [hunk 4]
```diff
@@ -840,6 +856,14 @@ The smallest denomination of [ether](#ether). 10<sup>18</sup> wei = 1 ether.
...
+
+<DocLink to="/developers/docs/scaling/layer-2-rollups/#zk-rollups">
+  Zero-knowledge Rollups
```
Comment (minimalsm) [inline src/content/glossary/index.md:864]:
```suggestion
  Zero-knowledge rollups
```

File: src/content/glossary/index.md [hunk 5]
```diff
@@ -713,6 +721,14 @@ Depositing a quantity of [ether](#ether) (your stake) to become a validator and
   Stake your ETH to become an Ethereum validator
 </DocLink>
 
+### STARK {#stark}
+
+Scalable Transparent ARgument of Knowledge a type of a [zero-knowledge proof](#zk-proof).
+
+<DocLink to="/developers/docs/scaling/layer-2-rollups/#zk-rollups">
+  Zero-knowledge Rollups
```
Comment (minimalsm) [inline src/content/glossary/index.md:729]:
```suggestion
  Zero-knowledge rollups
```

File: src/content/glossary/index.md [hunk 6]
```diff
@@ -681,6 +681,14 @@ A program that executes on the Ethereum computing infrastructure.
   Introduction to Smart Contracts
 </DocLink>
 
+### SNARK {#snark}
+
+Succinct Non-interactive ARgument of Knowledge a type of a [zero-knowledge proof](#zk-proof).
+
+<DocLink to="/developers/docs/scaling/layer-2-rollups/#zk-rollups">
+  Zero-knowledge Rollups
```
Comment (minimalsm) [inline src/content/glossary/index.md:689]: Unsure if we want this link in every kind-of related occurrence here. Anyone have any thoughts?

```suggestion
  Zero-knowledge rollups
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
Hey @mmilenkovic thanks for the PR here 💪

This looks really good. Just a few suggestions around language and site conventions.

File: src/content/glossary/index.md (hunk 3 above)
Comment (mmilenkovic) [reply src/content/glossary/index.md:861]: You are right. Maybe even change "individual" for "entity". But it is understandable as proposed.

**Comment by mmilenkovic** [comment]
@minimalsm Suggestions were spot on. All added.

File: src/content/glossary/index.md [hunk 7]
```diff
@@ -840,6 +856,14 @@ The smallest denomination of [ether](#ether). 10<sup>18</sup> wei = 1 ether.
 
 A special Ethereum address, composed entirely of zeros, that is specified as the destination address of a [contract creation transaction](#contract-creation-transaction).
 
+### Zero-knowledge proof {#zk-proof}
+
+Zero-knowledge proof is a cryptographic method that allows an individual to prove that a statement is true without conveying any additional information.
```
Comment (minimalsm) [inline src/content/glossary/index.md:861]:
```suggestion
A zero-knowledge proof is a cryptographic method that allows an individual to prove that a statement is true without conveying any additional information.
```

**Comment by minimalsm** [comment]
Thanks @mmilenkovic ❤️ Bringing this in now.

**Comment by minimalsm** [comment]
@all-contributors please add @mmilenkovic for docs, ideas

**Pull request merged**